protos/table.proto

```diff
@@ -320,6 +321,8 @@ message DataFile {
   //
   // When this is zero, it should be interpreted as "unknown".
   uint64 file_size_bytes = 6;
+  // The base path of the data file. Used when the file is imported or referred from another dataset
+  optional uint32 path_base_index = 7;
```

Comment:
nit: I think we can use a simpler name like `base` as long as the documentation is clear about what it means. 
nit2: When we say "Used when the file is imported or referred from another dataset", we should probably also mention that it should look up the actual base in `reference_paths`

Reply:
> oh actually, if you agree with using entries instead of a map in base paths, then we should call this `base_id` instead of `base_idx`

+1